File: python3/1678.py
class Solution:
    def interpret(self, c: str) -> str:
        r = []
        j = 0
        n = len(c)
        while j < n:
            if c[j] == "G":
                r.append("G")
                j += 1
            elif c[j + 1] == ")":
                r.append("o")
                j += 2
            else:
                r.append("a")
                r.append("l")
                j += 4
        return "".join(r)

File: javascript/1678.js
/**
 * @param {string} c
 * @return {string}
 */
var interpret = function (c) {
    let n = c.length, i = 0, r = "";
    for (let j = 0; j < n; j++) {
        if (c[j] == 'G')
            r += 'G';
        else if (c[j + 1] == ')') {
            r += 'o';
            j++;
        } else {
            r += "al";
            j += 3;
        }
    }
    return r;
};
